ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-answer/activityAnswer.component.ts

```diff
@@ -1,87 +1,108 @@
-                                      [class]="'file-answer-' + block.stableId"
-                                      [block]="block"
-                                      [readonly]="readonly"
-                                      [studyGuid]="studyGuid"
-                                      [activityGuid]="activityGuid"
-                                      (valueChanged)="onChange($event)"
-                                      (componentBusy)="componentBusy.next($event)">
-            </ddp-activity-file-answer>
-            <ddp-activity-matrix-answer *ngIf="isMatrixQuestion(block)"
-                                        [class]="'matrix-answer-' + block.stableId"
-                                        [block]="block"
-                                        [readonly]="readonly"
-                                        (valueChanged)="onChange($event)">
-            </ddp-activity-matrix-answer>
-            <ddp-activity-instance-select-answer *ngIf="isActivityInstanceSelectQuestion(block)"
-                                                 [class]="'activity-instance-select-answer-' + block.stableId"
-                                                 [block]="block"
-                                                 [readonly]="readonly"
-                                                 (valueChanged)="onChange($event)"
-                                                 (componentBusy)="componentBusy.next($event)"
-            ></ddp-activity-instance-select-answer>
-            <span *ngIf="block.additionalInfoFooter"
-                  [innerHTML]="block.additionalInfoFooter"
-                  class="ddp-activity-answer__info-footer">
-      </span>
-        </ng-container>`,
-    styleUrls: ['activityAnswer.component.scss']
+    template: ` <ng-container *ngIf="block.shown">
+        <ddp-activity-boolean-answer
+            *ngIf="isBooleanQuestion(block)"
+            [class]="'boolean-answer-' + block.stableId"
+            [block]="block"
+            [readonly]="readonly"
+            (valueChanged)="onChange($event)"
+        >
+        </ddp-activity-boolean-answer>
+        <ddp-activity-text-answer
+            *ngIf="isTextQuestion(block)"
+            [block]="block"
+            [readonly]="readonly"
+            (valueChanged)="onChange($event)"
+        >
+        </ddp-activity-text-answer>
+        <ddp-activity-numeric-answer
+            *ngIf="isNumericQuestion(block) || isDecimalQuestion(block)"
+            [class]="'numeric-answer-' + block.stableId"
+            [block]="block"
+            [readonly]="readonly"
+            (valueChanged)="onChange($event)"
+        >
+        </ddp-activity-numeric-answer>
+        <ddp-activity-picklist-answer
+            *ngIf="isPicklistQuestion(block)"
+            [class]="'picklist-answer-' + block.stableId"
+            [block]="block"
+            [readonly]="readonly"
+            [studyGuid]="studyGuid"
```

Comment:
hey @alextanlan I've added studyGuid and activityGuid but the weird thing is when the type is pickListQuestion those values just doesn't flow

Reply:
where it does not work for you ?
I have just checked on `ddp-basil` study locally.
There is a console message with the `studyGuid`, `activityGuid` (see screenshot below)
![image](https://user-images.githubusercontent.com/7396837/156217771-dd875237-4507-4131-a908-bf864bae9990.png)

Reply:
Oh wow, I was trying the pancan project on the join us